server/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -593,6 +603,32 @@ public void prepareForClose() {
         } catch (Exception e) {
             logger.warn("unexpected exception while waiting for http server to close", e);
         }
+
+        // Wait for async search tasks to finish. Since the HTTP server is closed, none of these tasks should be for synchronous searches
+        // even though the action name is the same.
+        var taskManager = injector.getInstance(TransportService.class).getTaskManager();
+        FutureTask<Void> searchAwaiter = new FutureTask<>(() -> {
+            boolean searchTasksFinished = true;
+            while (searchTasksFinished) {
```

Comment:
Should we use some minimal backoff?

Reply:
Probably. Is this the one time we actually get to use `Thread.sleep()`, given this will be running on its own thread?

Reply:
Heh, I dunno, maybe active polling is better, but it seems quite aggressive/expensive, especially since we use streams here. I really meant this as a question, I don't know what is best.